spec.html

```diff
@@ -12406,7 +12406,7 @@ <h1>Execution</h1>
       </emu-alg>
 
       <emu-note>
-        <p>Together with the definition of liveness, this clause prescribes legal optimizations that an implementation may apply regarding WeakRefs.</p>
+        <p>Together with the definition of liveness, this clause prescribes optimizations that an implementation may apply regarding WeakRefs.</p>
```

Comment:
When describing optimization prescription or proscription, "legal" is pretty widely used and understood vernacular. Can you expand on why you think it's not precise?

Reply:
I don't understand the inclusion concerns point, but I'm also fine with other single word descriptors like "conformant" or "valid" or "allowed" when describing optimizations. I do think a word like "available" actually makes it less precise.

Reply:
I don't agree that "legal" strictly refers to matters of law; games have legal/illegal moves, for example.

Reply:
> games have legal/illegal moves, for example

That is precisely the kind of **loose** linguistic extension that I referred to above; it does not change the **strict** "_of or pertaining to the law_" denotation.

Regardless, I have now changed the text of the note in [Shared Memory Guidelines](https://tc39.es/ecma262/multipage/memory-model.html#sec-shared-memory-guidelines) to use "allowed" rather than "legal" for better alignment with notes in e.g. [WeakRef/FinalizationRegistry Execution](https://tc39.es/ecma262/multipage/executable-code-and-execution-contexts.html#sec-weakref-execution) ("_optimizations are thus **allowed** to observably empty WeakRefs_") and [Sequentially Consistent Atomics](https://tc39.es/ecma262/multipage/memory-model.html#sec-memory-order) ("_[events] that are not constrained by happens-before or synchronizes-with are **allowed** to occur anywhere in the order_").